### DISCUSSION TITLE
Performance Disruption: Lag Caused by 'set' Method Impacts User Experience

### DISCUSSION THREAD

**Discussion by 4i8** [pmndrs/zustand#1990] [Bug report]
## Summary
When I use the `set` method in React, it causes significant lag, which can be quite disruptive to the user experience. This delay hampers the overall responsiveness of the application and can lead to frustration for users who expect smooth and seamless interactions. The noticeable slowdown in performance can potentially deter users from engaging with the application and may even impact its usability in critical scenarios. Therefore, addressing this lag issue with the set method is essential to ensure a satisfactory and efficient user experience.
## Link to reproduction

[57665bb2-e3a4-423a-a78e-1dd0e8c3ec34.webm](https://github.com/pmndrs/zustand/assets/76555623/1fc962ea-6df8-4d7b-a40f-b69f912595aa)

## Check List

Please do not ask questions in issues.

- [ ] I've already opened a [discussion](https://github.com/pmndrs/zustand/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [ ] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

Please check this if you're filing an issue regarding TypeScript.

- [ ] I've read the [typescript guide](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md), in particular that `create` is to be used as `create<T>()(...)` and not `create<T>(...)`.

**Comment by dai-shi** [comment]
Can you create a minimal reproduction with https://codesandbox.io/s/react-new ?

**Reply by 4i8** [reply]
Nerver Mind, now everything is working. Thank you, dai-shi, for this masterpiece of art. Simple is simple

**Reply by boblitex** [reply]
@4i8 curious as to what caused your lag and how you fixed it
